sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/DownloadCBDE.targets

```diff
@@ -9,79 +9,93 @@
-    <Message Importance="high" Text="CBDE: Updating CBDE binaries" Condition="$(UpdateCBDE)=='true'" />
-
-    <CallTarget Targets="DownloadFromRepox" Condition="$(ShouldDownloadCBDE)=='true'" />    
-    <!-- Specifies the CBDE files that should be included in the project and copied to the output directory -->
-    <ItemGroup>
-      <Content Include="$(RelativeCBDEFolder)\**\*.exe" >
-        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-      </Content>
-      <Content Include="$(RelativeCBDEFolder)\**\*.txt" >
-        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-      </Content>
-    </ItemGroup>
-  </Target>
-
-  <Target Name="DownloadFromRepox">
-    
-    <Message Importance="high" Text="CBDE: Deleting any existing files..." />
+    <Message Importance="high" Text="%(CBDEUrls.SubFolder)/%(CBDEUrls.MainExe): binaries exist? $(CBDEFilesExist)" />
+    <Message Importance="high" Text="%(CBDEUrls.SubFolder)/%(CBDEUrls.MainExe): Updating CBDE binaries" Condition="$(UpdateCBDE)=='true'" />
+    <Message Importance="high" Text="CBDE: Deleting any existing files..." Condition="$(ShouldDownloadCBDE)=='true'"/>
 
     <ItemGroup>
       <ExistingCBDEFiles Include="$(CBDEFolder)\*.*" />
     </ItemGroup>
-    <Delete Files="@(ExistingCBDEFiles)" />
+    <Delete Files="@(ExistingCBDEFiles)" Condition="$(ShouldDownloadCBDE)=='true'"/>
 
-    <PropertyGroup>
-      <PowerShellExe Condition="'$(PowerShellExe)'==''">%WINDIR%\System32\WindowsPowerShell\v1.0\powershell.exe</PowerShellExe>
-      <UnzipScript>$(MSBuildProjectDirectory)\..\..\..\Scripts\unzip.ps1</UnzipScript>
-      <DownloadScript>$(MSBuildProjectDirectory)\..\..\..\Scripts\download.ps1</DownloadScript>
-    </PropertyGroup>
-
-    <Message Importance="high" Text="CBDE: Begin downloading CBDE zip..." />
-    <Exec ConsoleToMsBuild="true"
-      Command="$(PowerShellExe) -NonInteractive -ExecutionPolicy Unrestricted -File $(DownloadScript) -url $(CBDEDownloadUrl) -outfolder $(CBDEFolder) -outfile $(CBDEZipFileName)">
-      <Output TaskParameter="ConsoleOutput" PropertyName="OutputOfExec" />
+    <Message Importance="high" Text="CBDE: Begin downloading CBDE zip..." Condition="$(ShouldDownloadCBDE)=='true'"/>
+    <Exec ConsoleToMsBuild="true" Condition="$(ShouldDownloadCBDE)=='true'"
+      Command="$(PowerShellExe) -NonInteractive -ExecutionPolicy Unrestricted -File $(DownloadScript) -url @(CBDEUrls) -outfolder $(CBDEFolder)%(CBDEUrls.SubFolder) -outfile %(CBDEUrls.ZipFileName)">
+      <Output TaskParameter="ConsoleOutput" PropertyName="OutputOfExec"/>
     </Exec>
-    <Message Importance="low" Text="Output of powershell download: $(OutputOfExec)" />
-    <Message Importance="high" Text="CBDE: Finished downloading CBDE binaries: $(CBDEZipFilePath)" />
+    <Message Importance="low" Text="Output of powershell download: $(OutputOfExec)"  Condition="$(ShouldDownloadCBDE)=='true'"/>
```

Comment:
It should be possible to use batching to avoid repeating this condition everywhere, but batching can be a pain to get working properly so if this works I'd leave it as it is.